src/XmlStringStreamer/Stream/File.php

```diff
@@ -13,27 +13,41 @@ class File implements StreamInterface
     public function __construct($mixed, $chunkSize = 16384, $chunkCallback = null)
     {
         if (is_string($mixed)) {
-            // Treat as filename
-            if (!file_exists($mixed)) {
-                throw new \Exception("File '$mixed' doesn't exist");
+            // Account for common stream/URL wrappers before checking if a file exists
+            $realPath = $mixed;
+            if (preg_match('/^([\w.]+):\/\//', $realPath, $matched)) {
+                if (preg_match('/(http|ftp|php|data|ssh2)/', $matched[1])) {
+                    // Disable file_exists() check
+                    $realPath = null;
+                } else {
+                    // Remove wrapper for file_exists() check.
+                    $realPath = substr($realPath, strlen($matched[0]));
+                }
+                switch ($matched[1]) {
+                    default:
+                        $filePath = null;
+                }
```

Comment:
I guess this is a remnant from something?

Reply:
Yep! Thanks for catching. Left over from my first pass at handling different wrappers before simplifying with the regex approach.